src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy

```diff
@@ -194,10 +215,41 @@ class FlowRulesSpec extends HealthCheckSpecification {
                                        DeleteRulesAction.OVERWRITE_DEFAULTS]) {
             expectedRules = (expectedRules + extraIngressFlowRule)
         }
+        if (s42IsEnabled && data.deleteRulesAction in [DeleteRulesAction.DROP_ALL_ADD_DEFAULTS,
+                                                       DeleteRulesAction.IGNORE_DEFAULTS,
+                                                       DeleteRulesAction.REMOVE_DEFAULTS,
+                                                       DeleteRulesAction.OVERWRITE_DEFAULTS]) {
+            /*in case: REMOVE_DEFAULTS and REMOVE_ADD_DEFAULTS
+            the 'expectedRules' variable is already includes extraS42FlowRttInputRule,
+            it is returned by the 'getFlowRules' method. 'getFlowRules' is not modified.
+            'extraS42FlowRttInputRule' is the flow rule and default rule at the same time */
+            expectedRules = (expectedRules + extraS42FlowRttInputRule)
+            if (data.deleteRulesAction == DeleteRulesAction.REMOVE_DEFAULTS) {
+                expectedRules = (expectedRules - extraS42FlowRttInputRule)
+            }
+        }
+
         def deletedRules = northbound.deleteSwitchRules(srcSwitch.dpId, data.deleteRulesAction)
 
         then: "The corresponding rules are really deleted"
-        deletedRules.size() == data.rulesDeleted
+        def amountOfDeletedRules = data.rulesDeleted
+        if (s42IsEnabled) {
+            if (northbound.getSwitchProperties(srcSwitch.dpId).server42FlowRtt) {
+                if (data.deleteRulesAction in [DeleteRulesAction.DROP_ALL,
+                                               DeleteRulesAction.DROP_ALL_ADD_DEFAULTS]) {
+                    amountOfDeletedRules += s42FlowRttInput + s42QinqOuterVlanCount + s42FlowRttIngressForwardCount
+                }
+                if (data.deleteRulesAction in [DeleteRulesAction.IGNORE_DEFAULTS,
+                                               DeleteRulesAction.OVERWRITE_DEFAULTS]) {
+                    amountOfDeletedRules += s42QinqOuterVlanCount + s42FlowRttIngressForwardCount
+                }
+                if (data.deleteRulesAction in [DeleteRulesAction.REMOVE_DEFAULTS,
+                                               DeleteRulesAction.REMOVE_ADD_DEFAULTS]) {
+                    amountOfDeletedRules += s42FlowRttInput
+                }
+            }
+        }
```

Comment:
This does not look good. Update `getExpectedRules` method accordingly for each required action